x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/api/util/update_rules.ts

```diff
@@ -40,3 +41,19 @@ export const convertEsqlQueryToTranslationResult = (
     ? RuleMigrationTranslationResultEnum.full
     : RuleMigrationTranslationResultEnum.partial;
 };
+
+export const transformRuleMigrationToUpdate = (
+  ruleMigration: UpdateRuleMigrationData
+): UpdateRuleMigrationData => {
+  const {
+    translation_result: translationResult,
+    elastic_rule: elasticRule,
+    ...rest
+  } = ruleMigration;
+  return {
+    ...rest,
+    elastic_rule: elasticRule,
+    translation_result:
+      translationResult ?? convertEsqlQueryToTranslationResult(elasticRule?.query),
+  };
+};
```

Comment:
This logic always sets the `translation_result`, even if we are not updating the `query`. If we receive an update only for let's say, the `title`, this will set `translation_result: undefined`, will this end up updating the field? It is a bit confusing.
What about this:

```suggestion
export const transformRuleMigrationToUpdate = (
  ruleMigration: UpdateRuleMigrationData
): UpdateRuleMigrationData => {
  if (ruleMigration.elasticRule?.query == null || ruleMigration.translation_result != null) {
    return ruleMigration;
  }
  return {
    ...ruleMigration,
    translation_result: convertEsqlQueryToTranslationResult(ruleMigration.elasticRule.query),
  };
};
```

Reply:
`convertEsqlQueryToTranslationResult(elasticRule?.query)` returns `undefined` if query is not defined. But if you think it is not clear in this context we can also adjust code here

Reply:
Yes, that's what I mean, the `translation_result: undefined` part is created and passed to the ES update operation. I know it is ignored by the ES API, but that is extra knowledge about the ES API that a future dev will need to have in order to understand this code.
Also, it would be better if the `query` was a required parameter of `convertEsqlQueryToTranslationResult`.

Reply:
done in https://github.com/elastic/kibana/pull/206285/commits/7d0a5c28ef60ef2a46cf090f897e35a83a7186d6